gradle-plugin/kubernetes/src/test/java/org/eclipse/jkube/gradle/plugin/GradleUtilTest.java

```diff
@@ -76,12 +76,28 @@ public void extractProperties_withComplexMap_shouldReturnValidProperties() {
     final JavaProject result = convertGradleProject(project);
     // Then
     assertThat(result.getProperties())
-        .hasSize(2)
+        .isNotEmpty()
         .hasFieldOrPropertyWithValue("object.field1", 1)
         .hasFieldOrPropertyWithValue("property.1", null)
         .hasEntrySatisfying("property.1", e -> assertThat(e).isEqualTo("test"));
   }
 
+  @Test
+  public void extractProperties_shouldContainSystemProperties() {
+    // Given
+    System.setProperty("foo.property", "somevalue");
+    when(project.getProperties()).thenReturn(Collections.emptyMap());
+
+    // When
+    final JavaProject result = convertGradleProject(project);
+
+    // Then
+    assertThat(result.getProperties())
+      .hasFieldOrPropertyWithValue("foo.property", null)
+      .hasEntrySatisfying("foo.property", e -> assertThat(e).isEqualTo("somevalue"));
+    System.clearProperty("foo.property");
```

Comment:
Use SystemMock to be able to provide a reliable test for this.

We can also add a test for property precedence.